code/game/objects/items/devices/uplinks.dm

```diff
@@ -187,12 +187,12 @@ A list of items and costs is stored under the datum of every game mode, alongsid
 			return
 		var/amount = input("How many telecrystals do you wish to withdraw?:", "Extract telecrystals", null) as num
 		if (amount > uses)
-			return
```

Comment:
this is fine, but just for the future you can oneline this with `amount = min(amount,uses)` which gets the lower of the two values

Reply:
definitively true, I forgot about that